test/import-map-validation.test.js

```diff
@@ -0,0 +1,46 @@
+const { findUrlsToValidateInServices } = require("../src/verify-valid-url");
+
+describe(`/import-map.json - services validation`, () => {
+  it(`finds all urls that have to be validated.`, async () => {
+    const mockImportMap = {
+      imports: {
+        a: "https://cdn.com/a-1.mjs",
+        b: "https://cdn.com/b-1.mjs",
+        c: "https://cdn.com/c-1.mjs",
+        "https://cdn.com/d-1.mjs": "https://cdn.com/d-update.mjs",
+      },
+    };
+
+    const toBeVerifiedUrls = findUrlsToValidateInServices(
+      mockImportMap.imports
+    );
+
+    expect(toBeVerifiedUrls.sort()).toEqual(
+      [
+        "https://cdn.com/a-1.mjs",
+        "https://cdn.com/b-1.mjs",
+        "https://cdn.com/c-1.mjs",
+        "https://cdn.com/d-update.mjs",
+      ].sort()
+    );
+  });
+
+  it(`does not validate relative urls.`, async () => {
+    const mockImportMap = {
+      imports: {
+        "/scope2/": "https://cdn.com/a-2.mjs",
```

Comment:
This is in an invalid entry in the import map. The value of an import with a trailing slashes must have a trailing slash. See section (2)(6) of https://wicg.github.io/import-maps/#sort-and-normalize-a-specifier-map.

We should add validation logic to the source code for this, so that users can't modify their import map such that it is invalid.

Reply:
that makes sense!